File: EzPay/src/main/java/com/thinhtran/EzPay/service/TransactionService.java
package com.thinhtran.EzPay.service;

import com.thinhtran.EzPay.dto.request.TopUpRequest;
import com.thinhtran.EzPay.dto.request.TransferRequest;
import com.thinhtran.EzPay.dto.response.StatisticsResponse;
import com.thinhtran.EzPay.dto.response.TransactionResponse;

import java.util.List;

public interface TransactionService {
    void transfer(String senderUsername, TransferRequest request);

    List<TransactionResponse> getHistory(String username);

    void topUp(TopUpRequest request);

    StatisticsResponse getStatistics();
}

File: frontend/src/services/transactionService.ts
import api from './api';
import { TransferRequest, Transaction, Statistics, TopUpRequest } from '../types';

export const transactionService = {
  transfer: async (transferData: TransferRequest): Promise<{ message: string }> => {
    const response = await api.post('/transactions', transferData);
    return response.data; // This returns the ApiResponse object with message
  },

  getHistory: async (): Promise<Transaction[]> => {
    const response = await api.get('/transactions');
    return response.data.data; // Access the data field within the ApiResponse
  },

  topUp: async (topUpData: TopUpRequest): Promise<{ message: string }> => {
    const response = await api.post('/transactions/top-up', topUpData);
    return response.data; // This returns the ApiResponse object with message
  },

  getStatistics: async (): Promise<Statistics> => {
    const response = await api.get('/transactions/statistics');
    return response.data.data; // Access the data field within the ApiResponse
  },
}; 